uvloop/includes/fork_handler.h

```diff
@@ -0,0 +1,15 @@
+
+typedef void (*OnForkHandler)();
+
+OnForkHandler __forkHandler = NULL;
+
+/* Auxiliary function to call global fork handler if defined.
+
+Note: Fork handler needs to be in C (not cython) otherwise it would require
+GIL to be present, but some forks can exec non-python processes.
```

Comment:
I'm not sure I follow... The old handler had a `nogil` decoration, i.e. `cdef void __atfork_child() nogil:`.  How's this indirection makes that different?

Reply:
There is small but significant difference in case you compile it with cython and nogil traces enabled (-DCYTHON_TRACE=1 -DCYTHON_TRACE_NOGIL=1). 
In this mode cython will insert trace calls during compilation and resulting compiled __atfork_child function will contain `__Pyx_TraceCall`. It is not a problem when fork contains python runtime, but it will freeze at this trace call in case forking child does not contains python runtime.

Traces are enabled in `make debug`, so I decide to write it in pure C where no such traces are inserted by cython.

Reply:
Just so that I understand the scope: this PR is only needed to unbreak `make debug`?

Reply:
No, on the contrary, this PR should fix freezing uvloop (under some conditions) and freezing occurs even when compiled with pure make (without `debug`), but for a different reason.

Reply:
Alright, I'll review the PR soon.

In the meantime, could you please squash all commits into 1 and write a commit message that explains the bug/problem in a detailed way?  It would help me / future maintainers a lot.

Huge props for coming up with a functional test case.

Reply:
Thanks, I squash it. Give me a note if there is something not clear enough.